spec/suse/connect/api_spec.rb

```diff
@@ -2,125 +2,166 @@
-      .with(headers: expected_request_headers)
-      .to_return(
-        status: 200,
-        body: [ { endpoint: 'organizations/orders' } ].to_json,
-        headers: {}
-      )
-
-    stub_request(:get, 'https://scc.suse.com/connect/organizations/products?page=1')
-      .with(headers: expected_request_headers)
-      .to_return(
-        status: 200,
-        body: [ { endpoint: 'organizations/products' } ].to_json,
-        headers: {}
-      )
-
-    stub_request(:get, 'https://scc.suse.com/connect/organizations/products/unscoped?page=1')
-      .with(headers: expected_request_headers)
-      .to_return(
-        status: 200,
-        body: [ { endpoint: 'organizations/products/unscoped' } ].to_json,
-        headers: {}
-      )
-
-    stub_request(:get, 'https://scc.suse.com/connect/organizations/repositories?page=1')
-      .with(headers: expected_request_headers)
-      .to_return(
-        status: 200,
-        body: [ { endpoint: 'organizations/repositories' } ].to_json,
-        headers: {}
-      )
-
-    stub_request(:get, 'https://scc.suse.com/connect/organizations/subscriptions?page=1')
-      .with(headers: expected_request_headers)
-      .to_return(
-        status: 200,
-        body: [ { endpoint: 'organizations/subscriptions' } ].to_json,
-        headers: {}
-      )
-  end
-
-  let(:url) { 'http://example.com' }
   let(:username) { 'scc_user' }
   let(:password) { 'scc_password' }
   let(:api_client) { described_class.new(username, password) }
-  let(:expected_request_headers) do
-    {
-      'Authorization' => 'Basic ' + Base64.encode64("#{username}:#{password}").strip,
-      'User-Agent' => "RMT/#{RMT::VERSION}"
-    }
-  end
-  let(:response_data) { { foo: 'bar' } }
-
-  describe '#make_single_request' do
-    subject { api_client.send(:make_single_request, 'GET', 'http://example.org/api_method') }
-
-    it { is_expected.to eq(response_data) }
-  end
-
-  describe '#make_paginated_request' do
-    subject { api_client.send(:make_paginated_request, 'GET', 'http://example.org/api_method') }
-
-    it { is_expected.to eq([response_data, response_data]) }
-  end
-
-  describe '#list_orders' do
-    subject { api_client.list_orders }
-
-    it { is_expected.to eq([ { endpoint: 'organizations/orders' } ]) }
+  let(:uuid) { 'test-uuid' }
+
+  describe '#system_uuid' do
+    subject(:method_call) { api_client.send(:system_uuid) }
+
+    before do
+      allow(SecureRandom).to receive(:uuid).and_return(uuid)
+    end
+
+    context 'when system_uuid file exists' do
+      before do
+        allow(File).to receive(:exist?).with(described_class::UUID_FILE_LOCATION).and_return(true)
+      end
+
+      it 'reads a file' do
+        expect(File).not_to receive(:write).with(described_class::UUID_FILE_LOCATION, uuid)
+        expect(File).to receive(:read).with(described_class::UUID_FILE_LOCATION).and_return(uuid)
+        expect_any_instance_of(described_class).to receive(:system_uuid).and_call_original
```

Comment:
Hmm? Why does the spec for `system_uuid` expect that `system_uuid` will be called? Also, why call the original implementation?